pkg/controller/network/routetable/controller.go

```diff
@@ -142,99 +158,134 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1beta1.RouteTable)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.SetConditions(runtimev1alpha1.Creating())
-	cr.UpdateExternalStatus(*rsp.RouteTable)
+	response, err := e.client.DescribeRouteTablesRequest(&awsec2.DescribeRouteTablesInput{
+		RouteTableIds: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
 
-	if err := e.createRoutes(ctx, meta.GetExternalName(cr), cr.Spec.Routes, cr.Status.Routes); err != nil {
-		return managed.ExternalCreation{}, err
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(ec2.IsRouteTableNotFoundErr, err), errDescribe)
 	}
 
-	if err := e.createAssociations(ctx, meta.GetExternalName(cr), cr.Spec.Associations, cr.Status.Associations); err != nil {
-		return managed.ExternalCreation{}, err
+	if response.RouteTables == nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateNotFound)
 	}
 
-	return managed.ExternalCreation{}, nil
-}
+	table := response.RouteTables[0]
+
+	patch, err := ec2.CreateRTPatch(table, cr.Spec.ForProvider)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
 
-func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
-	// TODO(soorena776): add more sophisticated Update logic, once we
-	// categorize immutable vs mutable fields (see #727)
+	if len(patch.Tags) != 0 {
+		// tagging the RouteTable
+		if _, err := e.client.CreateTagsRequest(&awsec2.CreateTagsInput{
+			Resources: []string{meta.GetExternalName(cr)},
+			Tags:      v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
+		}).Send(ctx); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errCreateTags)
+		}
+	}
+
+	if patch.Routes != nil {
+		// Attach the routes in Spec
+		if err := e.createRoutes(ctx, meta.GetExternalName(cr), cr.Spec.ForProvider.Routes, cr.Status.AtProvider.Routes); err != nil {
+			return managed.ExternalUpdate{}, err
+		}
+	}
+
+	if patch.Associations != nil {
+		// Associate route table to subnets in Spec.
+		if err := e.createAssociations(ctx, meta.GetExternalName(cr), cr.Spec.ForProvider.Associations, cr.Status.AtProvider.Associations); err != nil {
+			return managed.ExternalUpdate{}, err
+		}
+	}
 
 	return managed.ExternalUpdate{}, nil
 }
 
 func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
-	cr, ok := mgd.(*v1alpha3.RouteTable)
+	cr, ok := mgd.(*v1beta1.RouteTable)
 	if !ok {
 		return errors.New(errUnexpectedObject)
 	}
 
 	cr.Status.SetConditions(runtimev1alpha1.Deleting())
 
-	// in order to delete a route table, all of its dependencies need to be
-	// deleted first
-
-	if err := e.deleteRoutes(ctx, meta.GetExternalName(cr), cr.Status.Routes); err != nil {
-		return err
-	}
-
-	if err := e.deleteAssociations(ctx, cr.Status.Associations); err != nil {
```

Comment:
Do we not actually need to delete the routes?

Reply:
I verified this again. Existing routes do not restrict deletion of the route table. Only associations should be deleted before deleting the route table.